### DISCUSSION TITLE
Run OCSYank after copy relative path

### DISCUSSION THREAD

**Discussion by pshvechikov** [nvim-tree/nvim-tree.lua#1957] [Q&A]
The default command for copy into system clipboard via OSC symbols is 
`autocmd TextYankPost * if v:event.operator is 'y' && v:event.regname is '' | execute 'OSCYankReg "' | endif`

This command is not respected when one copies relative path (`Y`) from nvim-tree, since no TextYankPost is triggered with vim.fn.setreg: 

```
local function copy_to_clipboard(content)
  if M.use_system_clipboard == true then
    vim.fn.setreg("+", content)
    vim.fn.setreg('"', content)
    return notify.info(string.format("Copied %s to system clipboard!", content))
  else
    vim.fn.setreg('"', content)
    vim.fn.setreg("1", content)
    return notify.info(string.format("Copied %s to neovim clipboard!", content))
  end
end
```

How could one detect the change of the "+" register and autocmd the OSCYankReg command?

**Comment by alex-courtis** [comment]
Today I leaned of the existence of `TextYankPost`. It definitely does not fire on setreg and appears explicitly designed just for text yank.

Could we use an nvim-tree event, say, `Event.ClipboardCopied`, passing the text that was copied?

**Reply by pshvechikov** [reply]
That would be wonderful if such an event existed. Almost always paths (shortcuts `Y` or `gy`) are needed outside the vim. And it really sucks to paste from + register and yank the pasted text manually.

**Reply by alex-courtis** [reply]
> That would be wonderful if such an event existed. Almost always paths (shortcuts Y or gy) are needed outside the vim. And it really sucks to paste from + register and yank the pasted text manually.

I'm not quite understanding: why do you need to yank the pasted `+`? Is that to fire the OSC plugin?

We can add the event: #2312

[Pull Requests](https://github.com/nvim-tree/nvim-tree.lua/wiki/Development) are gratefully appreciated.

**Reply by pshvechikov** [reply]
> I'm not quite understanding: why do you need to yank the pasted +? Is that to fire the OSC plugin?

yes, you are right.

**Comment by Akmadan23** [comment]
I opened a PR (#2489) that should solve this issue. @pshvechikov can you please confirm if this is what you're after?
